### ISSUE TITLE
Setting br-instance name (ietf-softwire) with snabb config results in cltable error

### ISSUE THREAD

**Issue by takikawa** [Igalia/snabbswitch#701]
Example:

```
$ sudo ./snabb config set -s ietf-softwire 23739 "/softwire-config/binding/br/br-instances/br-instance[id=1]/name" "foo"
error: lib/cltable.lua:37: attempt to index local 'cltable' (a nil value)
```

Using `snabb config add` produces a more reasonable error:

```
$ sudo ./snabb config add -s ietf-softwire 29527 "/softwire-config/binding/br/br-instances/br-instance[id=1]/name" "foo"
apps/config/support/snabb-softwire-v1.lua:417: unsupported path: /softwire-config/binding/br/br-instances/br-instance[id=1]/name
```

**Comment by tsyesika** [comment]
There seems to be two issues here.

The first issue seems to be that we do have named programs in snabb and there is no reason why we can't change the name of them. We did however opt not to put the name in our config (`snabb-softwire-v1`) because we wanted a single configuration file to be usable with multiple lwaftr instances that might run at the same times (and names having a requirement of being unique). With that said I don't see why we couldn't have a non-mandatory field where a name could be specified or could be omitted and specified on the command line (as it is now). We then could allow this to be reconfigurable and actually have this succeed in the sense of actually setting or changing the name.

The second issue is obviously that it gives an obtuse error message which isn't only nonsensical but also not even consistent over the config tools. I think we need some code which checks for a properties existence over anything else and displays a good reasonable error. I think it'd be really good if we could catch all none-existent or supported fields across the different tools when doing translations and give a sensible error for them.

I think fixing the second of the two issues brought up should resolve this bug. I'd like to see the first issue solved at some point if no one sees an issue with it.

**Comment by tsyesika** [comment]
@takikawa Hey, I'm actually not able to reproduce this error on origin/lwaftr. I can't see why it wouldn't still be a bug but are you still able to reproduce it?

I get:
```
$ sudo ./snabb config set -s ietf-softwire lwaftr /softwire-config/binding/br/br-instances/br-instance[id=1]/name "foo"
apps/config/support/snabb-softwire-v1.lua:294: unrecognized leaf: name
```

**Comment by takikawa** [comment]
@xray7224 I just tried it on snabb1 on an up-to-date checkout and get the same behavior that you got.

I did a bisect to see when it changed, and it looks like the current behavior happens as of this commit: 6845bcab9fb76b94671c3537264460b683086d96

**Comment by dpino** [comment]
@xray7224  @takikawa It is "/softwire-config/binding/br/br-instances/br-instance[id=1]/name" a valid path according to the schema? In case it is, then the error message is wrong and I think the problem is in this if condition which is too generic:

```lua
-- Handle special br attributes (tunnel-payload-mtu, tunnel-path-mru, softwire-num-threshold)
if #path > #br_instance_paths then
   ...
   error('unrecognized leaf: '..path[#path].name)
end
```

**Comment by dpino** [comment]
The issue can be closed as it's solved now:

```bash
$ sudo ./snabb config set -s ietf-softwire 16253 "/softwire-config/binding/br/br-instances/br-instance[id=1]/name" "foo"

$ sudo ./snabb config get -s ietf-softwire 16253 "/softwire-config/binding/br/br-instances/br-instance[id=1]/name"
foo
```

Solved in #737.
